framework/PostProcessors/Metric.py

```
@@ -0,0 +1,289 @@
+      @ Out, None
+    """
+    if output.isOpen():
+      output.close()
+    if self.dynamic:
+      outputInstance = Files.returnInstance('DynamicXMLOutput', self)
+    else:
+      outputInstance = Files.returnInstance('StaticXMLOutput', self)
+    outputInstance.initialize(output.getFilename(), self.messageHandler, path=output.getPath())
+    outputInstance.newTree('MetricPostProcessor', pivotParam=self.pivotParameter)
+    outputResults = [outputDictionary] if not self.dynamic else outputDictionary.values()
+    for ts, outputDict in enumerate(outputResults):
+      pivotVal = outputDictionary.keys()[ts]
+      for nodeName, nodeValues in outputDict.items():
+        for metricName, value in nodeValues.items():
+          if type(value) == float:
+            outputInstance.addScalar(nodeName, metricName, value, pivotVal=pivotVal)
+          elif type(value) in [list, np.ndarray]:
+            if len(list(value)) == 1:
+              outputInstance.addScalar(nodeName, metricName, value[0], pivotVal=pivotVal)
+            else:
+              self.raiseAnError(IOError, "Multiple values are returned from metric '", metricName, "', this is currently not allowed")
+          elif type(value) == dict:
+            ## FIXME: The following are used to accept timedependent data, and should be checked later.
+            outputInstance.addVector(nodeName, metricName, value, pivotVal=pivotVal)
+          else:
+            self.raiseAnError(IOError, "Unrecognized type of input value '", type(value), "'")
+    outputInstance.writeFile()
+
+  def _writeText(self,output,outputDictionary, separator=' '):
+    """
+      Defines the method for writing the post-processor to a .csv file
+      @ In, output, File object, file to write to
+      @ In, outputDictionary, dict, dictionary stores importance ranking outputs
+      @ In, separator, string, optional, separator string
+      @ Out, None
+    """
+    ##TODO: Add this method if needed
+    pass
```

If this is not implemented, it would be better to throw an error than let it pass silently. (Otherwise you mystify the user as to why their text output is not written)